src/query/mlt/mlt.rs

```diff
@@ -0,0 +1,387 @@
+                        let term = Term::from_field_u64(field, val);
+                        *term_frequencies.entry(term).or_insert(0) += 1;
+                    }
+                }
+            }
+            FieldType::Date(_) => {
+                for field_value in field_values {
+                    // TODO: Ask if this is the semantic (timestamp) we want
+                    let val = field_value
+                        .value()
+                        .date_value()
+                        .ok_or(TantivyError::InvalidArgument("invalid value".to_string()))?
+                        .timestamp();
+                    if !self.is_noise_word(val.to_string()) {
+                        let term = Term::from_field_i64(field, val);
+                        *term_frequencies.entry(term).or_insert(0) += 1;
+                    }
+                }
+            }
+            FieldType::I64(_) => {
+                for field_value in field_values {
+                    let val = field_value
+                        .value()
+                        .i64_value()
+                        .ok_or(TantivyError::InvalidArgument("invalid value".to_string()))?;
+                    if !self.is_noise_word(val.to_string()) {
+                        let term = Term::from_field_i64(field, val);
+                        *term_frequencies.entry(term).or_insert(0) += 1;
+                    }
+                }
+            }
+            FieldType::F64(_) => {
+                for field_value in field_values {
+                    let val = field_value
+                        .value()
+                        .f64_value()
+                        .ok_or(TantivyError::InvalidArgument("invalid value".to_string()))?;
+                    if !self.is_noise_word(val.to_string()) {
+                        let term = Term::from_field_f64(field, val);
+                        *term_frequencies.entry(term).or_insert(0) += 1;
+                    }
+                }
+            }
+            _ => {}
+        }
+        Ok(())
+    }
+
+    /// Determines if the term is likely to be of interest based on "more-like-this" settings
+    fn is_noise_word(&self, word: String) -> bool {
+        let word_length = word.len();
+        if word_length == 0 {
+            return true;
+        }
+        if self
+            .min_word_length
+            .map(|min| word_length < min)
+            .unwrap_or(false)
+        {
+            return true;
+        }
+        if self
+            .max_word_length
+            .map(|max| word_length > max)
+            .unwrap_or(false)
+        {
+            return true;
+        }
+        return self.stop_words.contains(&word);
+    }
+
+    /// Couputes the score for each term while ignoring not useful terms
+    fn create_score_term(
+        &self,
+        searcher: &Searcher,
+        per_field_term_frequencies: HashMap<Term, usize>,
+    ) -> Result<Vec<ScoreTerm>> {
+        let mut score_terms = BinaryHeap::new();
```

Comment:
Why use a binary heap here?

It would have actually made sense if we were effectively using it to keep the Top-K, but we are not, so we might as well just use a Vec and sort it at the end. It would be clearer and more efficient.

The most efficient solution is to indeed use a binary heap, but then you need to keep its size below  max_query_terms